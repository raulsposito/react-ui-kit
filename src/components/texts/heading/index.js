import React from 'react'
import PropTypes from 'prop-types'

import MyPropTyes from '@global/propTypes'

import { H1, H2, H3, H4, H5, H6 } from './styles'

const Heading = ({ weight, size, text, type, align, color, children, marginBottom, disabled }) => {
  const Tag = (() => {
    switch (type) {
      case 'h2': return H2
      case 'h3': return H3
      case 'h4': return H4
      case 'h5': return H5
      case 'h6': return H6
      default: return H1
    }
  })()

  return (
    <Tag
      weight={weight}
      color={color}
      align={align}
      size={size}
      disabled={disabled}
      marginBottom={marginBottom}
    >
      {text || children}
    </Tag>
  )
}

Heading.propTypes = {
  text: PropTypes.string,
  type: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6']),
  align: PropTypes.oneOf(['left', 'center', 'right']),
  color: MyPropTyes.color,
  weight: MyPropTyes.weight,
  size: MyPropTyes.size,
  children: PropTypes.node,
  marginBottom: MyPropTyes.space,
  disabled: PropTypes.bool
}

export default Heading
