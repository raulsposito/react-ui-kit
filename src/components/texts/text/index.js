import React from 'react'
import PropTypes from 'prop-types'

import MyPropTyes from '@global/propTypes'

import { StyledText } from './styles'

const Text = ({ text, align, weight, color, size, children, marginBottom, disabled }) => (
  <StyledText weight={weight} size={size} color={color} align={align} marginBottom={marginBottom} disabled={disabled}>
    {text || children}
  </StyledText>
)

Text.propTypes = {
  text: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]),
  align: PropTypes.oneOf(['left', 'center', 'right']),
  weight: MyPropTyes.weight,
  color: MyPropTyes.color,
  size: MyPropTyes.size,
  children: PropTypes.node,
  marginBottom: MyPropTyes.space,
  disabled: PropTypes.bool
}

export default Text
