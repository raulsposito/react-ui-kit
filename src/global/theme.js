// https://www.color-hex.com/color/8293ee

export const colors = {
  primary: {
    regular: '#FC760D',
    dark: '#E46B0B',
    darker: '#C66F2B',
    hover: '#E17622',
    focus: '#B97641',
    selected: '#D4762C',
    disabled: '#F99546'
  },
  variant: {
    regular: '#FCB10D',
    dark: '#E29F0C',
    darker: '#C9A557',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  secondary: {
    regular: '#2FCFCF',
    dark: '#1F9A9A',
    darker: '#147676',
    hover: '#3DBDBD',
    focus: '#52A2A2',
    selected: '#44B4B4',
    disabled: '#5FD8D8'
  },
  success: {
    regular: '#3DB265',
    dark: '#48A668',
    darker: '#59946D',
    hover: '#48A668',
    focus: '#59946D',
    selected: '#59946D',
    disabled: '#9EBCA8'
  },
  error: {
    regular: '#FF5668',
    dark: '#E45C6B',
    darker: '#BA666F',
    hover: '#E45C6B',
    focus: '#BA666F',
    selected: '#BA666F',
    disabled: '#F8A4AC'
  },
  warning: {
    regular: '#FFAF00',
    dark: '#E5BB4D',
    darker: '#CCA644',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  contrast: {
    regular: '#3C3C3C',
    disabled: '#E4E4E4'
  },
  system: {
    regular: '#BAB5B4',
    dark: '#CBC6C4',
    darker: '#D7D4D1',
    hover: '#ACA8A8',
    focus: '#989595',
    selected: '#A6A2A1',
    disabled: '#C8C4C3'
  },
  background: {
    primary: '#FFFFFF',
    regular: '#FFFFFF',
    hover: '#F1F1F1',
    disabled: '#F8F8F8'
  },
  events: {
    inprogress: '#3DB265',
    next: '#5678CF',
    finished: '#BAB5B4',
    suspended: '#FFD056',
    cancelled: '#FF5668',
    white: '#FFFFFF'
  }
}

export const spacing = {
  quarter: '2px',
  half: '4px',
  threeQuarters: '6px',
  one: '8px',
  oneAndAQuarter: '10px',
  oneAndAHalf: '12px',
  two: '16px',
  twoAndAQuarter: '18px',
  twoAndAHalf: '20px',
  three: '24px',
  threeAndAHalf: '28px',
  four: '32px',
  fourAndAHalf: '36px',
  five: '40px',
  fiveAndAHalf: '44px',
  six: '48px',
  sixAndAHalf: '52px',
  seven: '56px',
  sevenAndAHalf: '60px',
  eight: '64px',
  eightAndAHalf: '68px',
  nine: '72px',
  nineAndAHalf: '76px',
  ten: '80px',
  tenAndAHalf: '84px',
  eleven: '88px',
  elevenAndAHalf: '92px',
  twelve: '96px',
  twelveAndAHalf: '100px',
  thirteen: '104px',
  fourteen: '112x',
  fifteen: '120px'
}

export const page = {
  width: '1200px'
}
