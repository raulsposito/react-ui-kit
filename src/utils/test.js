export const getComponents = styled => {
  const componentClass = styled.styledComponentId
  return document.getElementsByClassName(componentClass)
}

export const getComponent = (styled, index = 0) => {
  const components = getComponents(styled)
  return components[index]
}
