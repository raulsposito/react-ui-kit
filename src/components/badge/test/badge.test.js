import React from 'react'
import { render, screen } from '@testing-library/react'

import { colors, spacing } from '@global/theme'

import { getComponent } from '@utils/test'

import { Text } from '@components/texts'
import Icon from '@components/icon'

import Badge from '..'

describe('<Badge />', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Badge text='Luke Skywalker' />)
    expect(asFragment()).toMatchSnapshot()
  })

  describe('Text and Icon', () => {
    it('has only text', () => {
      render(<Badge text='Luke Skywalker' />)
      const t = screen.queryByText('Luke Skywalker')
      const i = getComponent(Icon)

      expect(t).toBeInTheDocument()
      expect(i).toBeUndefined()
    })

    it('has only icon', () => {
      render(<Badge icon='view_in_ar' />)
      const t = getComponent(Text)
      const i = screen.queryByText('view_in_ar')

      expect(t).toBeUndefined()
      expect(i).toBeInTheDocument()
    })

    it('has both', () => {
      const { container } = render(<Badge text='Luke Skywalker' icon='view_in_ar' />)
      const t = screen.getByText('Luke Skywalker')
      const i = screen.getByText('view_in_ar')

      expect(t).toBeInTheDocument()
      expect(i).toBeInTheDocument()

      const { childNodes } = container.firstChild
      expect(childNodes[0].textContent).toEqual('Luke Skywalker')
      expect(childNodes[1].textContent).toEqual('view_in_ar')
    })

    it('has both in reverse order', () => {
      const { container } = render(<Badge text='Luke Skywalker' icon='view_in_ar' reverse />)
      const t = screen.getByText('Luke Skywalker')
      const i = screen.getByText('view_in_ar')

      expect(t).toBeInTheDocument()
      expect(i).toBeInTheDocument()

      const { childNodes } = container.firstChild
      expect(childNodes[0].textContent).toEqual('view_in_ar')
      expect(childNodes[1].textContent).toEqual('Luke Skywalker')
    })
  })

  describe('Size', () => {
    it('is medium by default', () => {
      render(<Badge text='Skywalker' />)
      const b = screen.getByText('Skywalker').parentElement
      expect(b).toHaveStyle({ padding: `${spacing.one} ${spacing.oneAndAHalf}` })
    })

    it('is small', () => {
      render(<Badge text='Skywalker' size='small' />)
      const b = screen.getByText('Skywalker').parentElement
      expect(b).toHaveStyle({ padding: `${spacing.half}` })
    })

    it('is medium', () => {
      render(<Badge text='Skywalker' size='medium' />)
      const b = screen.getByText('Skywalker').parentElement
      expect(b).toHaveStyle({ padding: `${spacing.one} ${spacing.oneAndAHalf}` })
    })

    it('is large', () => {
      render(<Badge text='Skywalker' size='large' />)
      const b = screen.getByText('Skywalker').parentElement
      expect(b).toHaveStyle({ padding: `${spacing.oneAndAQuarter} ${spacing.two}` })
    })
  })

  describe('Colors & Types', () => {
    describe('Solid', () => {
      describe('Regular colors', () => {
        it('is primary by default', () => {
          render(<Badge text='Skywalker' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ 'background-color': `${colors.primary.regular}` })
          expect(t).toHaveStyle({ color: `${colors.background.regular}` })
        })

        it('is primary', () => {
          render(<Badge text='Skywalker' color='primary' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ 'background-color': `${colors.primary.regular}` })
          expect(t).toHaveStyle({ color: `${colors.background.regular}` })
        })

        it('is secondary', () => {
          render(<Badge text='Skywalker' color='secondary' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ 'background-color': `${colors.secondary.regular}` })
          expect(t).toHaveStyle({ color: `${colors.background.regular}` })
        })

        it('is variant', () => {
          render(<Badge text='Skywalker' color='variant' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ 'background-color': `${colors.variant.regular}` })
          expect(t).toHaveStyle({ color: `${colors.background.regular}` })
        })

        it('is success', () => {
          render(<Badge text='Skywalker' color='success' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ 'background-color': `${colors.success.regular}` })
          expect(t).toHaveStyle({ color: `${colors.background.regular}` })
        })

        it('is warning', () => {
          render(<Badge text='Skywalker' color='warning' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ 'background-color': `${colors.warning.regular}` })
          expect(t).toHaveStyle({ color: `${colors.background.regular}` })
        })

        it('is error', () => {
          render(<Badge text='Skywalker' color='error' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ 'background-color': `${colors.error.regular}` })
          expect(t).toHaveStyle({ color: `${colors.background.regular}` })
        })

        it('is system', () => {
          render(<Badge text='Skywalker' color='system' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ 'background-color': `${colors.system.regular}` })
          expect(t).toHaveStyle({ color: `${colors.background.regular}` })
        })
      })

      describe('Event colors', () => {
        it('is inprogress', () => {
          render(<Badge text='Skywalker' color='inprogress' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ 'background-color': `${colors.events.inprogress}` })
          expect(t).toHaveStyle({ color: `${colors.background.regular}` })
        })

        it('is next', () => {
          render(<Badge text='Skywalker' color='next' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ 'background-color': `${colors.events.next}` })
          expect(t).toHaveStyle({ color: `${colors.background.regular}` })
        })

        it('is finished', () => {
          render(<Badge text='Skywalker' color='finished' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ 'background-color': `${colors.events.finished}` })
          expect(t).toHaveStyle({ color: `${colors.background.regular}` })
        })

        it('is suspended', () => {
          render(<Badge text='Skywalker' color='suspended' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ 'background-color': `${colors.events.suspended}` })
          expect(t).toHaveStyle({ color: `${colors.background.regular}` })
        })

        it('is cancelled', () => {
          render(<Badge text='Skywalker' color='cancelled' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ 'background-color': `${colors.events.cancelled}` })
          expect(t).toHaveStyle({ color: `${colors.background.regular}` })
        })

        it('is white', () => {
          render(<Badge text='Skywalker' color='white' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ 'background-color': `${colors.events.white}` })
          expect(t).toHaveStyle({ color: `${colors.contrast.regular}` })
        })
      })
    })

    describe('Line', () => {
      describe('Regular colors', () => {
        it('is primary by default', () => {
          render(<Badge text='Skywalker' type='line' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ border: `2px solid ${colors.primary.regular}` })
          expect(t).toHaveStyle({ color: `${colors.primary.regular}` })
        })

        it('is primary', () => {
          render(<Badge text='Skywalker' color='primary' type='line' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ border: `2px solid ${colors.primary.regular}` })
          expect(t).toHaveStyle({ color: `${colors.primary.regular}` })
        })

        it('is secondary', () => {
          render(<Badge text='Skywalker' color='secondary' type='line' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ border: `2px solid ${colors.secondary.regular}` })
          expect(t).toHaveStyle({ color: `${colors.secondary.regular}` })
        })

        it('is variant', () => {
          render(<Badge text='Skywalker' color='variant' type='line' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ border: `2px solid ${colors.variant.regular}` })
          expect(t).toHaveStyle({ color: `${colors.variant.regular}` })
        })

        it('is success', () => {
          render(<Badge text='Skywalker' color='success' type='line' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ border: `2px solid ${colors.success.regular}` })
          expect(t).toHaveStyle({ color: `${colors.success.regular}` })
        })

        it('is warning', () => {
          render(<Badge text='Skywalker' color='warning' type='line' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ border: `2px solid ${colors.warning.regular}` })
          expect(t).toHaveStyle({ color: `${colors.warning.regular}` })
        })

        it('is error', () => {
          render(<Badge text='Skywalker' color='error' type='line' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ border: `2px solid ${colors.error.regular}` })
          expect(t).toHaveStyle({ color: `${colors.error.regular}` })
        })

        it('is system', () => {
          render(<Badge text='Skywalker' color='system' type='line' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ border: `2px solid ${colors.system.regular}` })
          expect(t).toHaveStyle({ color: `${colors.system.regular}` })
        })
      })

      describe('Event colors', () => {
        it('is inprogress', () => {
          render(<Badge text='Skywalker' color='inprogress' type='line' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ border: `2px solid ${colors.events.inprogress}` })
          expect(t).toHaveStyle({ color: `${colors.events.inprogress}` })
        })

        it('is next', () => {
          render(<Badge text='Skywalker' color='next' type='line' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ border: `2px solid ${colors.events.next}` })
          expect(t).toHaveStyle({ color: `${colors.events.next}` })
        })

        it('is finished', () => {
          render(<Badge text='Skywalker' color='finished' type='line' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ border: `2px solid ${colors.events.finished}` })
          expect(t).toHaveStyle({ color: `${colors.events.finished}` })
        })

        it('is suspended', () => {
          render(<Badge text='Skywalker' color='suspended' type='line' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ border: `2px solid ${colors.events.suspended}` })
          expect(t).toHaveStyle({ color: `${colors.events.suspended}` })
        })

        it('is cancelled', () => {
          render(<Badge text='Skywalker' color='cancelled' type='line' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ border: `2px solid ${colors.events.cancelled}` })
          expect(t).toHaveStyle({ color: `${colors.events.cancelled}` })
        })

        it('is white', () => {
          render(<Badge text='Skywalker' color='white' type='line' />)
          const t = screen.getByText('Skywalker')
          const b = t.parentElement
          expect(b).toHaveStyle({ border: `2px solid ${colors.events.white}` })
          expect(t).toHaveStyle({ color: `${colors.events.white}` })
        })
      })
    })
  })
})
