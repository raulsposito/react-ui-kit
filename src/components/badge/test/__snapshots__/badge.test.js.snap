// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-fubCzh bDVRMz"
  >
    <p
      class="sc-bdfBQB gZQgqc"
      color="background"
    >
      Luke Skywalker
    </p>
  </div>
</DocumentFragment>
`;
